### DISCUSSION TITLE
Trying to improve both Photoshop CC 2021 / 2022

### DISCUSSION THREAD

**Discussion by artik** [LinSoftWin/Photoshop-CC2022-Linux#33] [Ideas]
Hi @MiMillieuh. I'm happy to see I'm not the only one trying to get Photoshop running as good as on Windows. With CC 2021, I did it, but almost with 2022. In my case I have:

- `CC 2022` running with GPU acceleration since couple of months (but I'm facing sync bug).
- `CC 2021` **running perfectly without any sync issue and full GPU acceleration** (but with max Nvidia 470 drivers, after that (510, etc) my photoshop crashes.

I noticed that in your video, you don't have a proper `GPU acceleration` even if in preferences the "GPU acceleration" checkbox is checked. (you can check that in the font of the splash screen, yours are transparent, mine are in proper grey) or with the "srubby zoom" (top icon is greyed) that is not working.

In my videos I made a **scrubbly zoom** demo to show you. Also, not all CC 2021 are compatible as adobe has a bug. You must use CC 2021 22.1.1. Right after, with version 2021 22.5.2, I'm facing the same issur as reposrted abose with 2022 (Sync issue).

Here are my videos (I'm on a 180hz screen, but videos are 60hz):
`CC 2022` https://www.youtube.com/watch?v=G2YZ2wqT4Gk
`CC 2021` https://www.youtube.com/watch?v=FpgVUUnCRZA

Maybe we can join our effort and definetly make this Photoshop (2021 or 2022) running on latest drivers. I'm also trying to compile wine myself with patches to try fixing this sync issue.

**Comment by MiMillieuh** [comment]
Hello, 

Thanks for your report.

Maybe are Nvidia drivers a bit easier to get working with wine.
I have a lot of issue due to the open nature of AMD driver and a lot of library aren't installed with the base driver. (Also my GPU is a potato, just an integrated AMD gpu on a laptop)

Also the CC 2021 version is : 22.1.1 20210104.r.138 2021/01/04: 852879028a7  x64

Actually I didn't really noticed issue except that on my hardware it's slower than on Windows (on CC 2021). and Well in CC 2022 GPU acceleration is broken due to VKchildwindow (adobe now use multiple window and it breaks compatibility with VKD3D) and even with a properly self compiled and patched wine this issue won't disappear. When working around it I just end up with other glitches and bugs (that's kinda funny sometimes)

CC 2022 is really hard to get working... basic functions will work but not everything will unlike CC 2021 that has almost everything working

Also if you're facing issue with 510 Nvidia drivers, I've heard that a lot of people had issue with them in a lot of software so maybe that's actually Nvidia's fault.

if you want to compile wine with a patch I know how to so don't bother about that I can handle it.

If you have any suggestions I'm open to them.

Thanks for your contribution.

**Comment by artik** [comment]
Amazing, the version of wine i'm using is with lutris, and the `wine-lutris-ge-lol-7.0-2-x86_64` wine https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/7.0-GE-2-LoL.

Where we should start is getting the latest wine/proton (7.18 ?) that includes these `LoL` patches https://github.com/GloriousEggroll/wine-ge-custom/tree/master/patches/LoL

Also, you said "even with a properly self compiled and patched wine this issue won't disappear", do you have a compiled version for me to try ? Curious to see how it goes with Nvidia GPU

**Comment by MiMillieuh** [comment]
Well I'm not sure Lol patches should be used for this... Maybe I'm wrong after all but I didn't considerate it. also some patches require a specific wine version.

If you want to try my wine versions, they are here : https://lulucloud.mywire.org/FileHosting/GithubProjects/CustomWine/wine-tkg-staging-fsync-selfcompilated.zip

3 separated folder for 3 versions with differents patches.

**Comment by artik** [comment]
Damn it, no luck

```
2022-05-25 17:06:52,021: Starting Lutris 0.5.10
2022-05-25 17:06:52,045: No folder at /home/artik/.local/share/lutris/runners/retroarch/
2022-05-25 17:06:52,078: Using NVIDIA drivers 510.73.05 for x86_64
2022-05-25 17:06:52,079: GPU: NVIDIA GeForce RTX 3080
2022-05-25 17:06:52,079: GPU: 10DE:2206 1043:87B2 (nvidia drivers)
2022-05-25 17:06:52,123: Startup complete
Started initial process 3038 from gamemoderun env DXVK_STATE_CACHE=0 DXVK_LOG_LEVEL=none /home/artik/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-git-VKpatch/bin/wine /home/artik/.wine/drive_c/Program Files/Adobe/Adobe Photoshop 2021 22.1.1/Photoshop.exe
Start monitoring process.
/home/artik/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-git-VKpatch/bin/wine: /lib/i386-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /home/artik/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-git-VKpatch/bin/wine)
Monitored process exited.
Initial process has exited (return code: 256)
All processes have quit
Exit with return code 256
2022-05-25 17:06:55,529: Game still running (state: running)
2022-05-25 17:06:55,530: Stopping Photoshop 2021 (wine)
2022-05-25 17:06:55,530: The game has run for a very short time, did it crash?
```

**Comment by MiMillieuh** [comment]
Does it works with 470 drivers ?

**Comment by artik** [comment]
Exactly the same issue `version 'GLIBC_2.34' not found`
I also tried to install and fsync patched kernel, same error

Strange thing is only the Lol version make `CC 2021` perfectly working (except I'm blocked in 470 drivers)

**Comment by MiMillieuh** [comment]
I know that Glibc has also issue sometime on Nvidia with Steam so maybe it's related. try to install gcc or something like that

Edit : Maybe you're missing the package glibc (you're on mint so ubuntu based) : `sudo apt install glibc`

**Comment by artik** [comment]
`E: Unable to locate package glibc`

And online version 2.34 is experimental https://tracker.debian.org/pkg/glibc. Things are getting more complexe

**Comment by MiMillieuh** [comment]
> `E: Unable to locate package glibc`
> 
> And online version 2.34 is experimental https://tracker.debian.org/pkg/glibc. Things are getting more complexe

Maybe there is a ppa out there.

That's weird because in my distro it isn't experimental... Maybe mint is using an old Ubuntu base. Is your distro up to date (Latest release too) ?

**Comment by artik** [comment]
Hum, totally up to date :)

```
System:    Host: linuxmint Kernel: 5.13.0-44-generic x86_64 bits: 64 compiler: N/A Desktop: Cinnamon 5.2.7 
           Distro: Linux Mint 20.3 Una base: Ubuntu 20.04 focal 
Machine:   Type: Desktop System: ASUS product: N/A v: N/A serial: <superuser/root required> 
           Mobo: ASUSTeK model: ROG MAXIMUS Z690 HERO v: Rev 1.xx serial: <superuser/root required> 
           UEFI: American Megatrends v: 1304 date: 03/07/2022 
Battery:   Device-1: hid-80:4a:14:6d:aa:d5-battery model: Magic Keyboard with Numeric Keypad charge: N/A 
           status: Discharging 
CPU:       Topology: 12-Core model: 12th Gen Intel Core i9-12900K bits: 64 type: MT MCP arch: N/A 
           L2 cache: 30.0 MiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 152985 
           Speed: 600 MHz min/max: 800/6500 MHz Core speeds (MHz): 1: 708 2: 1404 3: 600 4: 601 5: 701 
           6: 1745 7: 725 8: 601 9: 801 10: 800 11: 800 12: 800 13: 600 14: 800 15: 800 16: 800 17: 800 
           18: 600 19: 600 20: 600 21: 600 22: 600 23: 879 24: 700 
Graphics:  Device-1: NVIDIA vendor: ASUSTeK driver: nvidia v: 470.129.06 bus ID: 01:00.0 
           Display: x11 server: X.Org 1.20.13 driver: nvidia tty: N/A 
           OpenGL: renderer: NVIDIA GeForce RTX 3080/PCIe/SSE2 v: 4.6.0 NVIDIA 470.129.06 
```

Do you think we can try a simple compilation of latest `7.18` proton with `Lol` patches? Maybe patches will work as they patched already the `7.0.2 proton version`

Also, I'm still wondering why you don't have the `Scrubbly Zoom` working, there is also something wrong here that maybe we can fix

**Comment by MiMillieuh** [comment]
I can try. But I know why the LOL patch work for you...

Lol patches use an older glibc version (2.28). That's why it works

Maybe newer wine version require something more upstream

Do you think you could be able to use the 2.34  version ?

Through Deb or ppa ? 

Also Ubuntu 20 is 2 yo

**Comment by artik** [comment]
I doubt Wine will use an experimental version of `glibc`. I'll try to update anyway and I let you know

Edit: my bad, I just saw `A new upstream version is available: 2.35`. I'm on it

**Comment by artik** [comment]
All my tests were unsuccessull haha. Tried to compile `glibc 2.35` but I realised  the package I needed to install was `libc6_2.34-0ubuntu3.2_amd64.deb`. But I have a bad dependency, and needed `libfakeroot` in an earlier version. When `libfakeroot_1.25.3-1.1_amd64.deb` is installed, I still have the bad dependency error. What else can I try ?

**Comment by MiMillieuh** [comment]
> All my tests were unsuccessull haha. Tried to compile `glibc 2.35` but I realised the package I needed to install was `libc6_2.34-0ubuntu3.2_amd64.deb`. But I have a bad dependency, and needed `libfakeroot` in an earlier version. When `libfakeroot_1.25.3-1.1_amd64.deb` is installed, I still have the bad dependency error. What else can I try ?

Maybe you could try other driver... I'm not sure... You could also try in a live ISO

Does the wine shipped in your distro works ? if not maybe your should check if a fresh install wouldn't fix the issue (try in live iso first of course so you don't wipe anything)
